python/packages/anthropic/agent_framework_anthropic/_chat_client.py

```diff
@@ -0,0 +1,662 @@
+        if tools_and_mcp := self._convert_tools_to_anthropic_format(chat_options.tools):
+            run_options.update(tools_and_mcp)
+        if chat_options.additional_properties:
+            run_options.update(chat_options.additional_properties)
+        run_options.update(kwargs)
+        return run_options
+
+    def _convert_messages_to_anthropic_format(self, messages: MutableSequence[ChatMessage]) -> list[dict[str, Any]]:
+        """Convert a list of ChatMessages to the format expected by the Anthropic client.
+
+        This skips the first message if it is a system message,
+        as Anthropic expects system instructions as a separate parameter.
+        """
+        # first system message is passed as instructions
+        if messages[0].role == Role.SYSTEM:
+            return [self._convert_message_to_anthropic_format(msg) for msg in messages[1:]]
+        return [self._convert_message_to_anthropic_format(msg) for msg in messages]
+
+    def _convert_message_to_anthropic_format(self, message: ChatMessage) -> dict[str, Any]:
+        """Convert a ChatMessage to the format expected by the Anthropic client.
+
+        Args:
+            message: The ChatMessage to convert.
+
+        Returns:
+            A dictionary representing the message in Anthropic format.
+        """
+        a_content: list[dict[str, Any]] = []
+        for content in message.contents:
+            match content.type:
+                case "text":
+                    a_content.append({"type": "text", "text": content.text})
+                case "data":
+                    if content.has_top_level_media_type("image"):
+                        a_content.append({
+                            "type": "image",
+                            "source": {"data": content.uri, "media_type": content.media_type},
+                        })
+                case "uri":
+                    if content.has_top_level_media_type("image"):
+                        a_content.append({"type": "image", "source": {"type": "url", "url": content.uri}})
+                case "function_call":
+                    a_content.append({
+                        "type": "tool_use",
+                        "id": content.call_id,
+                        "name": content.name,
+                        "input": content.parse_arguments(),
+                    })
+                case "function_result":
+                    a_content.append({
+                        "type": "tool_result",
+                        "tool_use_id": content.call_id,
+                        "content": prepare_function_call_results(content.result),
+                        "is_error": content.exception is not None,
+                    })
+                case "text_reasoning":
+                    a_content.append({"type": "thinking", "thinking": content.text})
+                case _:
+                    logger.debug(f"Ignoring unsupported content type: {content.type} for now")
+
+        return {
```

Comment:
I remember anthropic has some restriction requiring user-assistant-user-assistant ordering and also requires system message to be the first message. Are we going to do some implicit transformation here?